aws/existing-vpc/main.tf

```diff
@@ -54,12 +54,27 @@ variable "admin_ips" {
   type        = "list"
 }
 
+variable "vpc_id" {
+  description = "Optionally provided VPC ID to choose"
+  default     = ""
+}
+
 ////////////////////////////////////////////
 /////////////// END VARIABLES //////////////
 ////////////////////////////////////////////
 
 provider "aws" {}
 
+// we use intermediate local variables. So whenever it is needed to replace
+// or drop a modules it is easier to change just the local variable instead
+// of all other references
+locals {
+  key_name     = "${aws_key_pair.deployer.key_name}"
+  vpc_id       = "${coalesce(var.vpc_id, data.aws_vpc.default.id)}"
```

Comment:
my intention was more to have users of this digging into the TF files and just change the data resource. IMO `data.aws_vpc.default.id` might fail on a legacy accound not having a default VPC so the suggestion would be letting the user put his vpc id into the data resource.

But lets keep your suggestion for now as I think users might like your provided way